### PR TITLE
Solving the FindBug violations.

### PR DESCRIPTION
**Pull request by sebing**
Fixed the FindBug issues with PrintWriter and OutputStreams encoding.

Fixing the issue(https://github.com/yegor256/takes/issues/574)

### CODE REVIEW

**Comment by sebing** [comment]
@davvd One of the Travis jobs, Java 7 is failed, I think this issue is related to (https://travis-ci.org/yegor256/takes/builds/108122499)

**Comment by sebing** [comment]
@davvd Can you find someone to review it?

**Comment by davvd** [comment]
@sebing I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@longtimeago review this one, please

File: pom.xml [hunk 1]
```diff
@@ -306,44 +306,80 @@
...
+                          PMD, findbugs and other types of violations.
                          -->
                         <version>0.15.4</version>
                         <configuration>
                             <excludes combine.children="append">
                                 <exclude>checkstyle:/src/site/resources/.*</exclude>
                                 <!--
-                                @todo #517:30min Continue fixing FindBugs violations started in #517.
-                                 #517 turned on the FindBugs, temporarily excluding some classes.
+<<<<<<< HEAD
```
Comment (longtimeago) [inline pom.xml]: @sebing looks like something wasn't merged properly.
This line should not be in pom

File: src/main/java/org/takes/facets/fallback/FbLog4j.java [hunk 2]
```diff
@@ -66,9 +67,11 @@ public FbLog4j() {
...
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
         final Throwable error = req.throwable();
-        final PrintWriter writer = new PrintWriter(baos);
-        error.printStackTrace(writer);
-        writer.close();
+        final PrintStream stream = new PrintStream(
+            baos, false, StandardCharsets.UTF_8.toString()
+        );
+        error.printStackTrace(stream);
+        stream.close();
```
Comment (longtimeago) [inline src/main/java/org/takes/facets/fallback/FbLog4j.java]: @sebing We should close the stream in finally block

File: src/main/java/org/takes/facets/fallback/FbSlf4j.java [hunk 3]
```diff
@@ -75,9 +76,11 @@ public FbSlf4j() {
...
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
         final Throwable error = req.throwable();
-        final PrintWriter writer = new PrintWriter(baos);
-        error.printStackTrace(writer);
-        writer.close();
+        final PrintStream stream = new PrintStream(
+            baos, false, StandardCharsets.UTF_8.toString()
+        );
+        error.printStackTrace(stream);
+        stream.close();
```
Comment (longtimeago) [inline src/main/java/org/takes/facets/fallback/FbSlf4j.java]: @sebing same here

File: pom.xml (hunk 1 above)
Comment (sebing) [inline pom.xml]: @longtimeago Corrected the mistake by removing them

File: src/main/java/org/takes/http/BkBasic.java [hunk 4]
```diff
@@ -117,12 +118,16 @@ private void print(final Request req, final OutputStream output)
...
+        throws IOException {
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        final PrintWriter writer = new PrintWriter(baos);
-        err.printStackTrace(writer);
-        writer.close();
+        final PrintStream stream = new PrintStream(
+            baos, false, StandardCharsets.UTF_8.toString()
+        );
+        err.printStackTrace(stream);
+        stream.close();
```
Comment (longtimeago) [inline src/main/java/org/takes/http/BkBasic.java]: @sebing and here

File: src/test/java/org/takes/tk/TkProxyTest.java [hunk 5]
```diff
@@ -43,37 +44,49 @@
...
- *  The tests should verify the different HTTP methods (GET, POST, etc),
- *  as well as the different combinations of request/response headers.
  */
 public final class TkProxyTest {
 
     /**
-     * TkProxy can work.
+     * An array of http methods for testing.
+     */
+    private static final String[] METHODS = {
```
Comment (longtimeago) [inline src/test/java/org/takes/tk/TkProxyTest.java]: @sebing we already have such constants in `org.takes.rq.RqMethod`

**Comment by longtimeago** [comment]
@sebing please, see several remarks above

File: src/main/java/org/takes/facets/fallback/FbLog4j.java (hunk 2 above)
Comment (sebing) [inline src/main/java/org/takes/facets/fallback/FbLog4j.java]: @longtimeago I corrected them, I will push them with other changes.

File: src/main/java/org/takes/facets/fallback/FbSlf4j.java (hunk 3 above)
Comment (sebing) [inline src/main/java/org/takes/facets/fallback/FbSlf4j.java]: @longtimeago I corrected them, I will push them with other changes.

File: src/main/java/org/takes/http/BkBasic.java (hunk 4 above)
Comment (sebing) [inline src/main/java/org/takes/http/BkBasic.java]: @longtimeago I corrected them, I will push them with other changes.

**Comment by sebing** [comment]
@longtimeago Thank you for the suggestions. I corrected them, Can you re-review them?

**Comment by longtimeago** [comment]
@rultor merge

**Comment by longtimeago** [comment]
@sebing thanks!

**Comment by sebing** [comment]
@yegor256 Can you confirm the change? so rultor can merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, help us to evaluate quality here, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@davvd Looks good!

**Comment by davvd** [comment]
> @davvd Looks good!

@elenavolokhova thank you, good quality is always good ;)

**Comment by davvd** [comment]
@longtimeago I added 10 mins to @elenavolokhova (for QA review) in transaction 78638660. **22 mins** sent to your balance (ID `78638672`), many thanks! It took 173 hours and 28 mins.. review comments (c=7) added as a bonus. +22 to your rating, your total score is [+6153](http://www.netbout.com/b/35024?open=rating)

**Pull request merged**